skytemple_files/graphics/chara_wan/writer.py

```diff
@@ -0,0 +1,1412 @@
+                              'MFDisabled', int(newFramePiece.isDisabled()))
+            # rotation and scaling - ALWAYS 1 when DISABLE is 0
+            # Y offset
+
+            # resolution x
+            # flip vertical
+            # flip horizontal
+            # last frame
+            # unused - leave 0
+            # X offset
+
+            # palette - this will be autocalculated
+            # priority - ALWAYS 3
+            # tileindex - ALWAYS follows the rules of memory placement: 4 blocks = +1 index.  1 block min
+
+            ##document the used config
+            metaFrameData.append(newFramePiece)
+            if newFramePiece.isLast():
+                break
+
+        wan.frameData.append(metaFrameData)
+
+    in_file.seek(ptrOffsetsTable)
+    wan.offsetData = []
+    for offset_idx in range(len(ptrMetaFrames)):
+        headX = int.from_bytes(in_file.read(2), 'little', signed=True)
+        headY = int.from_bytes(in_file.read(2), 'little', signed=True)
+        lhandX = int.from_bytes(in_file.read(2), 'little', signed=True)
+        lhandY = int.from_bytes(in_file.read(2), 'little', signed=True)
+        rhandX = int.from_bytes(in_file.read(2), 'little', signed=True)
+        rhandY = int.from_bytes(in_file.read(2), 'little', signed=True)
+        centerX = int.from_bytes(in_file.read(2), 'little', signed=True)
+        centerY = int.from_bytes(in_file.read(2), 'little', signed=True)
+        wan.offsetData.append(FrameOffset((headX, headY), (lhandX, lhandY), (rhandX, rhandY), (centerX, centerY)))
+
+    ##read all anim pointers
+    wan.animGroupData = []
+    for ptrAnimGroup in ptrAnimGroups:
+        animGroup = []
+        for a_idx, ptrAnim in enumerate(ptrAnimGroup):
+            # if repeating the same pointer, it's the same animation. skip
+            if a_idx > 0 and ptrAnim == ptrAnimGroup[a_idx - 1]:
+                continue
+            in_file.seek(ptrAnim)
+            animSequence = []
+            while True:
+                frameDur = int.from_bytes(in_file.read(1), 'little')
+                flag = int.from_bytes(in_file.read(1), 'little')
+                frameIndex = int.from_bytes(in_file.read(2), 'little')
+                sprOffX = int.from_bytes(in_file.read(2), 'little', signed=True)
+                sprOffY = int.from_bytes(in_file.read(2), 'little', signed=True)
+                sdwOffX = int.from_bytes(in_file.read(2), 'little', signed=True)
+                sdwOffY = int.from_bytes(in_file.read(2), 'little', signed=True)
+                if frameDur == 0:
+                    break
+                else:
+                    animSequence.append(SequenceFrame(frameIndex, frameDur, flag,
+                                                    (sprOffX, sprOffY), (sdwOffX, sdwOffY)))
+            animGroup.append(animSequence)
+        wan.animGroupData.append(animGroup)
+
+    return wan
+
+
+def ImportSheets(inDir):
```

Comment:
Please move that to a separate module. For portaits I named it `sprite_bot_sheet.py` but you can also come up with a better name :P